### PR TITLE
Replace hostname -f with unmae -n

### PR DESCRIPTION
**Pull request by nak3**
OpenShift should not depend on `hostname` command. 
From RHEL 7 hostnamectl is used for controlling the HostName. In addition, the specification of `hostname` is being changed by upstream often.

And then, in case `uname -n` (even if `hostname -f`) returns error, it means Fatal. I think it is impossible to return error by `uname -n` (or `hostname -f`).

### CODE REVIEW

**Comment by smarterclayton** [comment]
We are consistent with upstream.  Please work with them to switch that, then we can change this once upstream has changed.

File: pkg/cmd/server/start/node_args.go [hunk 1]
```diff
@@ -104,13 +99,13 @@ func (args NodeArgs) BuildSerializeableNodeConfig() (*configapi.NodeConfig, erro
...
-func defaultHostname() (string, error) {
+func defaultHostname() string {
 
 	// Note: We use exec here instead of os.Hostname() because we
 	// want the FQDN, and this is the easiest way to get it.
-	fqdn, err := exec.Command("hostname", "-f").Output()
+	fqdn, err := exec.Command("uname", "-n").Output()
 	if err != nil {
-		return "", fmt.Errorf("Couldn't determine hostname: %v", err)
+		glog.Fatalf("Couldn't determine hostname: %v", err)
```
Comment (smarterclayton) [inline pkg/cmd/server/start/node_args.go]: This should be handled by other code, not here.  We do not glog.Fatal in common code.

**Comment by nak3** [comment]
I think I couldn't tell you correctly. `hostname -f` is wrapper command of `uname -n`.

My concern is that it is not simple to get the Hostname by `hostname -f`. It will look up somewhere `/etc/hostname`, `/proc/sys/kernel/hostname`, `/etc/hosts` or `uname -n`. hostname's upstream is often updating it and it is a little difficult to follow their update.

On the other hand, `uname -n` is a front end for a system call and simply get hostanme. Please refer gethostname(2)  man page.

**Comment by smarterclayton** [comment]
That may be the case, but we're simply trying to fit with kube.  Can you open an upstream issues on Kubernetes about this with this info?  I just want to make sure we're consistent.

----- Original Message -----

**Comment by nak3** [comment]
Ah... I understand. Your upstream means Kubernetes. I thought hostname command's upstream.
Sure I will.

File: pkg/cmd/server/start/node_args.go (hunk 1 above)
Comment (nak3) [inline pkg/cmd/server/start/node_args.go]: Even upstream is using `glog.Fatalf` , we should not use it here?
[1] https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/node.go#L31-L34

Comment (smarterclayton) [inline pkg/cmd/server/start/node_args.go]: They are doing startup differently - we only use glog.Fatal in very specific places, and everywhere else is required to return an error.

**Comment by smarterclayton** [comment]
Fixed in upstream - https://github.com/GoogleCloudPlatform/kubernetes/pull/7072.  Can you update this pull against the code (we might have added hostname -f in a few places) and then I'll review and merge?

Thanks for following through on this.

**Comment by nak3** [comment]
Regarding to OpenShift, `hostname` which is getting fqdn is only here. I think I don't need any update from previous commit.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**